src/MetadataUtility/Metadata/FrontierLabs/FlacCommentExtractor.cs

```diff
@@ -0,0 +1,141 @@
+                    };
+
+                    if (microphone.Type != null && !microphone.Type.Equals(FrontierLabs.UnknownMicrophoneString))
+                    {
+                        microphones.Add(microphone);
+                    }
+
+                    micNumber++;
+                }
+
+                // Update recording information with parsed comments
+                recording = recording with
+                {
+                    Sensor = (recording.Sensor ?? new Sensor()) with
+                    {
+                        Firmware = recording.Sensor?.Firmware ?? (string)this.ParseComment(FrontierLabs.FirmwareCommentKey, comments),
+                        BatteryLevel = recording.Sensor?.BatteryLevel ?? (string)this.ParseComment(FrontierLabs.BatteryLevelCommentKey, comments),
+                        LastTimeSync = recording.Sensor?.LastTimeSync ?? (OffsetDateTime?)this.ParseComment(FrontierLabs.LastSyncCommentKey, comments),
+                        SerialNumber = recording.Sensor?.SerialNumber ?? (string)this.ParseComment(FrontierLabs.SensorIdCommentKey, comments),
+                        Microphones = recording.Sensor?.Microphones ?? new Microphone[microphones.Length()],
+                    },
+                    StartDate = recording.StartDate ?? (OffsetDateTime?)this.ParseComment(FrontierLabs.RecordingStartCommentKey, comments),
+                    EndDate = recording.EndDate ?? (OffsetDateTime?)this.ParseComment(FrontierLabs.RecordingEndCommentKey, comments),
+                    Location = (recording.Location ?? new Location()) with
+                    {
+                        Longitude = recording.Location?.Longitude ?? (location != null ? location[FrontierLabs.LongitudeKey] : null),
+                        Latitude = recording.Location?.Latitude ?? (location != null ? location[FrontierLabs.LatitudeKey] : null),
+                    },
+                    MemoryCard = (recording.MemoryCard ?? new MemoryCard()) with
+                    {
+                        ManufacturerID = recording.MemoryCard?.ManufacturerID ?? (sdCid != null ? (byte)sdCid[SdCardCid.ManufacturerIDKey] : null),
+                        OEMID = recording.MemoryCard?.OEMID ?? (sdCid != null ? (string)sdCid[SdCardCid.OEMIDKey] : null),
+                        ProductName = recording.MemoryCard?.ProductName ?? (sdCid != null ? (string)sdCid[SdCardCid.ProductNameKey] : null),
+                        ProductRevision = recording.MemoryCard?.ProductRevision ?? (sdCid != null ? (float)sdCid[SdCardCid.ProductRevisionKey] : null),
+                        SerialNumber = recording.MemoryCard?.SerialNumber ?? (sdCid != null ? (uint)sdCid[SdCardCid.SerialNumberKey] : null),
+                        ManufactureDate = recording.MemoryCard?.ManufactureDate ?? (sdCid != null ? (string)sdCid[SdCardCid.ManufactureDateKey] : null),
+                    },
+                };
```

Comment:
I think this is a big improvement. I've created a method between the recording and the comment parsing (ParseComment) to do two checks for us.

1. Make sure the comment exists
2. Make sure there are no errors while parsing (check the fin), logging them if there are

Let me know what you think.

Reply:
much cleaner, yeah